### PR TITLE
80179 [10-10 EZ] Use VANotify to send submission failure email

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- This work is behind a feature toggle (flipper): YES
- Implemented the EZ Submission Failure email to use VANotify. These failure emails are only sent when the user supplies an email address when filling out the form. It utilizes the ` VANotify::EmailJob` to retry any failed email sends. 
- Created a new service VA Notify called `1010 Health Apps` and a new template in there for our failure email.
- We will need to have the staging template moved to production in VANotify before we merge this. 
- 1010 Health Apps Team

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/80179

## Testing done

- [x] *New code is covered by unit tests*
-  Previously used the `ApplicationMailer`. 
- We will send a test email in a non prod environment, or inspect the VANotify email queue to ensure our email is there.

## Screenshots
Screenshot 2024-06-13 at 10 12 12 AM

## What areas of the site does it impact?
The only area impacted is a user receiving a failed submission error from the 1010 EZ form. 

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated ([link to documentation](https://github.com/department-of-veterans-affairs/va.gov-team/pull/85721))
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -1331,6 +1331,10 @@ vanotify:
       api_key: fake_secret
       template_id:
         pega_status_update_email_template_id: pega_status_update_email_template_id
+    health_apps_1010:
+      api_key: fake_secret
+      template_id:
+        form1010_ez_failure_email: fake_template_id
```
[the remark this answers is not in the document]
Comment (coope93) [reply config/settings.yml:1337]: These are stored in aws parameter store that follows the path set here. So for example the staging api_key is in `dsva-vagov/vets-api/staging/vanotify/services/health_apps_1010/api_key`. I had to open a ticket with the platform team to add them.

File: app/models/health_care_application.rb [hunk 2]
```diff
@@ -271,8 +271,19 @@ def log_submission_failure
...
+    api_key = Settings.vanotify.services.health_apps_1010.api_key
+
+    VANotify::EmailJob.perform_async(
+      email,
+      template_id,
+      nil,
+      api_key
+    )
+  rescue => e
+    Rails.logger.error "Failure sending EZ Submission Failure Email. Error: #{e.message}"
```
[the remark this answers is not in the document]
Comment (coope93) [reply app/models/health_care_application.rb:286]: Yeah that looks right; good catch! 400's do currently send a sentry log per the code in the VANotify email job, but that raises the exception for any other errors that it looks like we send to sentry vs the rails logger. I'll update it 👍 .

[the remark this answers is not in the document]
Comment (coope93) [reply app/models/health_care_application.rb:286]: @dellerbie Updated to log to Sentry

[the remark this answers is not in the document]
Comment (coope93) [reply app/models/health_care_application.rb]: I updated this to log to sentry now

File: app/models/health_care_application.rb [hunk 3]
```diff
@@ -271,8 +271,19 @@ def log_submission_failure
...
   end
 
-  def send_failure_mail
-    HCASubmissionFailureMailer.build(parsed_form['email'], google_analytics_client_id).deliver_now
+  def send_failure_email
+    email = parsed_form['email']
+    template_id = Settings.vanotify.services.health_apps_1010.template_id.form1010_ez_failure_email
+    api_key = Settings.vanotify.services.health_apps_1010.api_key
+
+    VANotify::EmailJob.perform_async(
```
Comment (coope93) [inline app/models/health_care_application.rb:279]: Hopefully using the job here will lower our chance of failures when sending the emails.

**Review by ryan-mcneil — DISMISSED** [review]
LGTM

**Review by ryan-mcneil — DISMISSED** [review]
👍

**Review by JoshingYou1 — APPROVED** [review]
LGTM!

**Review by ryan-mcneil — APPROVED** [review]
👍

**Pull request merged**